src/sentry/static/sentry/app/utils/withRelease.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import Reflux from 'reflux';
+import createReactClass from 'create-react-class';
+
+import {Client} from 'app/api';
+import {Deploy, Release} from 'app/types';
+import getDisplayName from 'app/utils/getDisplayName';
+import ReleaseStore, {getReleaseStoreKey} from 'app/stores/releaseStore';
+import {getRelease, getReleaseDeploys} from 'app/actionCreators/releases';
+
+type DependentProps = {
+  api: Client;
+  orgSlug: string;
+  projectSlug: string;
+  releaseVersion: string;
+};
+
+type InjectedProps = {
+  release?: Release;
+  releaseLoading?: boolean;
+  releaseError?: Error;
+  deploys?: Array<Deploy>;
+  deploysLoading?: boolean;
+  deploysError?: Error;
+};
+
+const withRelease = <P extends InjectedProps>(WrappedComponent: React.ComponentType<P>) =>
+  createReactClass<
+    Omit<P, keyof InjectedProps> & Partial<InjectedProps> & DependentProps,
+    InjectedProps
+  >({
+    displayName: `withRelease(${getDisplayName(WrappedComponent)})`,
+    mixins: [Reflux.listenTo(ReleaseStore, 'onStoreUpdate') as any],
+
+    getInitialState() {
+      const {projectSlug, releaseVersion} = this.props as P & DependentProps;
+      const releaseData = ReleaseStore.get(projectSlug, releaseVersion);
+      return {...releaseData};
+    },
+
+    componentDidMount() {
+      this.fetchRelease();
+      this.fetchDeploys();
+    },
+
+    fetchRelease() {
+      const {api, orgSlug, projectSlug, releaseVersion} = this.props as P &
+        DependentProps;
+      const releaseData = ReleaseStore.get(projectSlug, releaseVersion);
+
+      if (!releaseData.release && !releaseData.releaseLoading) {
+        // HACK(leedongwei): Actions fired by the ActionCreators are queued to
```

Comment:
How/where does the action creator fire twice?

Reply:
If a page uses 2+ components which binds with `withRelease`, then the actionCreator would be fired at least 2+ while rendering the page. This happens on the IssueDetails page.